packages/ketcher-react/src/script/editor/tool/sgroup.ts

```diff
@@ -547,7 +554,10 @@ function getContextBySgroup(restruct, sgAtoms) {
   return anyChainedBonds(sgBonds) ? SgContexts.Group : SgContexts.Bond
 }
 
-function getContextBySelection(restruct, selection) {
+function getContextBySelection(
+  restruct: { molecule: any },
+  selection: { atoms: Iterable<any> | null | undefined; bonds: any[] }
```

Comment:
dont think this is a good typecheck, maybe "any" would do for the time being until we come up with a good method to generalize it?

Reply:
Probably, from the **interface Selection**, it can be:
atoms?: Array<number>
bonds?: Array<number>

Reply:
Used Selection and ReStruct interfaces [3d8bd84](https://github.com/epam/ketcher/pull/2007/commits/3d8bd84dd99850aed4a48aa63121b366fe5d98c2)